### PR TITLE
ansible_freeipa_module: Fix comparison of bool parameters in compare_…

### PR DESCRIPTION
**Pull request by t-woerner**
…args_ipa

Bool types are not iterable. Therefore the comparison using sets was failing
with a TypeError. This prevented to change the bool parameters for hosts.

A test for the host module has been added to verify that the bool parameters
can be modified.

New test:

  tests/host/test_host_bool_params.yml

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1784514

### CODE REVIEW

File: tests/host/test_host_bool_params.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    ipahost:
+      ipaadmin_password: MyPassword123
+      name: "{{ host1_fqdn }}"
+      requires_pre_auth: yes
+      ok_as_delegate: yes
+      ok_to_auth_as_delegate: yes
+    register: result
+    failed_when: result.changed
+
+  - name: Host "{{ host1_fqdn }}" present with !requires_pre_auth, !ok_as_delegate and !ok_to_auth_as_delegate
```
Comment (varunmylaraiah) [inline tests/host/test_host_bool_params.yml]: is this absent?
"absent with !requires_pre_auth, !ok_as_delegate and !ok_to_auth_as_delegate"

Comment (t-woerner) [reply tests/host/test_host_bool_params.yml]: All bool params are changed there at once.. later one by one

Comment (t-woerner) [reply tests/host/test_host_bool_params.yml]: I fixed the names of the tasks

**Comment by varunmylaraiah** [comment]
LGTM

**Pull request merged**